src/lint/track_config.nim

```diff
@@ -89,6 +89,43 @@ proc hasValidOnlineEditor(data: JsonNode, path: string): bool =
     else:
       result = false
 
+proc isValidKeyFeature(data: JsonNode, context: string, path: string): bool =
+  if isObject(data, context, path):
+    result = true
+
+    # TODO: Enable the `icon` checks when we have a list of valid icons.
+    if false:
+      const icons = [
+        "todo",
+      ].toHashSet()
+
+      if checkString(data, "icon", path):
+        let s = data["icon"].getStr()
+
+        if s notin icons:
+          let msg = "The value of `key_features.icon` is `" & s &
+                    "` which is not one of the valid values"
+          result.setFalseAndPrint(msg, path)
```

Comment:
I prefer `if false` to `when false` here so that we ensure the code continues to compile.

And I think it's better to disable the `icon` check completely for now - if we instead set `isRequired = false` then we would currently produce errors for tracks that have the empty string as the value of `icon`.

Reply:
Agreed